packages/java/engine-core/src/main/java/dev/hilla/engine/GeneratorShellRunner.java

```diff
@@ -41,14 +35,22 @@ public void add(String... args) {
 
     public void run(String input) throws InterruptedException, IOException {
         Objects.requireNonNull(input);
+        List<String> args;
+
+        if (FrontendUtils.isWindows()) {
+            args = List.of("cmd.exe", "/c", arguments.stream()
+                    .map(arg -> (arg.contains(" ") ? "\"" + arg + "\"" : arg))
+                    .collect(Collectors.joining(" ", "\"", "\"")));
```

Comment:
These lines mean that there will be three arguments: the latest one is combined to a string. I would suggest combining all arguments in the single array like that:
```suggestion
            args = Stream.concat(Stream("cmd.exe", "/c"), arguments.stream()
                    .map(arg -> (arg.contains(" ") ? "\"" + arg + "\"" : arg)))
                    .collect(Collectors.toList());
```
In my opinion, it looks more correct.

Reply:
I do want three strings.

Reply:
Why?

Reply:
Because that's the fix: `cmd /c` works better with a single argument, otherwise I found no combination of escapes that works

Reply:
Oh, I see, thanks